builtin/mainmenu/common.lua

```diff
@@ -93,30 +72,28 @@ function render_serverlist_row(spec, is_favorite)
 		end
 	end
 
-	local grey_out = not is_server_protocol_compat(spec.proto_min, spec.proto_max)
+	local grey_out = not spec.is_compatible
 
-	local details
-	if is_favorite then
-		details = "1,"
-	else
-		details = "0,"
-	end
+	local details = {}
 
-	if spec.ping then
-		local ping = spec.ping * 1000
-		if ping <= 50 then
-			details = details .. "2,"
-		elseif ping <= 100 then
-			details = details .. "3,"
-		elseif ping <= 250 then
-			details = details .. "4,"
+	if spec.lag or spec.ping then
+		local lag = (spec.lag or 0) * 1000 + (spec.ping or 0) * 250
+		if lag <= 125 then
+			table.insert(details, "1")
+		elseif lag <= 175 then
+			table.insert(details, "2")
+		elseif lag <= 250 then
+			table.insert(details, "3")
 		else
-			details = details .. "5,"
+			table.insert(details, "4")
 		end
 	else
-		details = details .. "0,"
+		table.insert(details, "0")
 	end
 
+	table.insert(details, ",")
```

Comment:
I originally had it this way, but it doesn't matter - because `table.concat({"whatever", "", "", "whatever"}, ",") == table.concat({"whatever", ",", "whatever"}, ",") == ` - and this way, it should be better for performance.

Reply:
I know that, but for correctness I consider it better not to mix the separator into the real data just because it gets the right result.